docs/manual/src/internals/ffi_converter_traits.md

```diff
@@ -0,0 +1,68 @@
+For these types, we normally can't implement `Lift` in the code we generate in the crate since neither `uniffi::Lift` or `url::Url` is local to that crate.
+This same issue applies to all of the FFI converter traits.
+
+To work around this we:
+
+* Add a generic parameter to each trait (`Lift` becomes `Lift<UT>` where "UT" is short for `UniFfiTag`).
+* Define a unit struct in each crate named `UniFfiTag` (the term "tag" is borrowed from the [C++ template pattern](https://www.geeksforgeeks.org/tag-dispatch-in-cpp/)).
+* We use that unit struct as the generic parameter for the trait (e.g. `Lift<crate::UniFfiTag>` is used to lift a value).
+
+Using the local type as a generic parameter means the impl no longer violates the orphan rule.
+For details on this see the [Rust Chalk Book](https://rust-lang.github.io/chalk/book/clauses/coherence.html#the-orphan-rules-in-rustc)
+The TLDR is that generic parameters "count" towards the requirement that there be a local type in the impl.
+
+In essence, `Lift<crate::UniFfiTag>` is a crate-local version of the `Lift` trait which gets around the orphan rule.
+However, this makes it harder to use this impl from another crate.
+UniFFI handles that in 2 ways:
+
+* The `uniffi` crate generates blanket trait impls (`for <UT> impl Lift<UT> for String`).
+  This allows all crates to use them automatically with their `UniFfiTag` struct.
+* UniFFI defines the `use_remote_type!` macro, which generates an implementation for the local
+  `UniFfiTag` by forwarding to the implementation from another crate's `UniFfiTag`.
+  See the [Remote and external types](../types/remote_ext_types.md#remote-external-types) for example usage.
+  This is also what the `remote` flag of the [custom type macro](../types/custom_types.md) does.
+
+### An incomplete list of FFI traits
+
+UniFFI defines a large number of FFI conversion traits, each one used for a specific purpose.
+This section describes a few them for explanatory purposes.
+See `uniffi_core/src/ffi_converter_traits.rs` for a full and up-to-date list.
+
+* `Lift`: Lift an value
+* `Lower`: Lower a value
+* `LowerReturn`: Lower a return value.
+  * For most types this is equivalent `Lower`, but a specialized impl is created for `Return<T, E>`.
+* `LiftRef`: Lift for a reference type.
+  This is often just `Lift` then a borrow, but a specialized impl is created for `Arc<T>`.
+* `FfiConverter`: General-purpose FFI conversion logic.
+  When `FfiConverter` is defined on a type, all other FFI traits are automatically derived.
+  This is what we implement for user-defined types like records and enums.
+* `FfiConverterArc`: FfiConverter implementation for `Arc<T>`.
+  This is another trait that we use to get around orphan rules.
+  Crate's can't directly define `FfiConverter` on `Arc<T>` for some interface, so they define `FfiConverterArc` instead.
+  `uniffi` defines a blanket `for T: FfiConverterArc impl FfiConverter<Arc<T>> `
```

Comment:
Same syntax again.
As this is not valid Rust syntax as is, should we maybe explain that better?